### DISCUSSION TITLE
Help with setting up content to populate

### DISCUSSION THREAD

**Discussion by listguy** [pkosiec/mongo-seeding#156] [Q&A] [answered]
Does this library still maintained?

**Comment by pkosiec** [comment]
Yes it is, you can see new commits and releases on the repo. Do you have any issue? If so, please describe it throughtly.

**Comment by listguy** [comment]
Hey, yes I do have an issue. Just wanted to make sure I'm not putting effort describing the issue for nothing 😅
Anyway I was  earlier trying to seed my DB (I am working on [mongoDB atlas](https://www.mongodb.com/cloud/atlas)) with a pre made JSON file. I followed the instructions in [importing-data](https://github.com/pkosiec/mongo-seeding/tree/master/examples/import-data) and [imported data definition](https://github.com/pkosiec/mongo-seeding/blob/master/docs/import-data-definition.md) and this is what I came up with: 

`seeder.js`
```js
require("dotenv").config();
const { Seeder } = require("mongo-seeding");
const path = require("path");

const config = {
  database: process.env.MONGO_URI,
  dropDatabase: false,
};

const seeder = new Seeder(config);

const seed = seeder.readCollectionsFromPath(path.resolve("./seeds/tickets"), {
  transformers: [Seeder.Transformers.replaceDocumentIdWithUnderscoreId],
});

seeder
  .import(seed)
  .then((_) => console.log("seeded successfully"))
  .catch((err) => {
    console.log(err);
    console.log("error");
  });

```

`tickets.json`:
```
  [{
    // Array of dummy data objects
  }]
```

File structure is like so:
```
     .
   ├── server
   │    ├── seeds
   │    │   └── tickets
   │    │      └── tickets.json
   │    └── seeder.js
```

When I tried to run `seeder.js` the code executes and a "seeded successfully" message logged to the console.
However, the collection was still empty. I tried to debug and found out `seed` was just an empty array `[]`. 
Seems like there is a problem in `seeder.readCollectionsFromPath()`? (No error message was logged to the console)

PS. in the end I managed to dig into the `node_modules`  and follow the errors in the console, and came up with a workaround solution like so:

`seeder.js`:
```js
require("dotenv").config();
const { Seeder } = require("mongo-seeding");
const fs = require("fs");
const config = {
  database: process.env.MONGO_URI,
  dropDatabase: false,
};

const seeder = new Seeder(config);

// const seed = seeder.readCollectionsFromPath(path.resolve("./seeds/tickets"), {
//   transformers: [Seeder.Transformers.replaceDocumentIdWithUnderscoreId],
// });

const seed = fs.readFileSync("./tickets.json");

seeder
  .import([JSON.parse(seed)])
  .then((_) => console.log("seeded successfully!"))
  .catch((err) => {
    console.log(err);
    console.log("error");
  });

```

`tickets.json`:
```
   {
      "name": "tickets",
      "documents": [
        // mock data ...
      ]
    } 
```

**Comment by pkosiec** [comment, marked as answer]
From what I can see you have totally different content from what is described in the documentation [here](https://github.com/pkosiec/mongo-seeding/blob/master/docs/import-data-definition.md).

You wrote:
```javascript
seeder.readCollectionsFromPath(path.resolve("./seeds/tickets"), ...)
```

But in fact you should point to `./seeds`:
```javascript
seeder.readCollectionsFromPath(path.resolve("./seeds"), ...)
```

Collections are represented by directories. Also, the `tickets.json` has invalid content. See the document I linked above to fix it.

Closing it for now as it works as described in docs. If it won't work, please write a comment here and I can reopen it 🙂 However, please double check you followed the official guides first. Cheers!

**Reply by pkosiec** [reply]
I converted the issue to a discussion, as this is proper place for such discussions. In few days I will delete the original issue completely. Of course the whole discussion will be archived here 🙂

**Reply by listguy** [reply]
Man! I can't believe it was a typo! 😂
I read the documentation closely a few times and still missed the directories path thing. The solution really was just to change `"./seeds/tickets"` to `"./seeds"`.
It would be nice if there was some kind of a warning for cases like mine (maybe an error to the console). If not, this discussion will do the trick 😄

Works as expected now, thank you for your time and for this awesome library 🙌

**Reply by pkosiec** [reply]
Glad it worked 👍 However, please keep in mind that the content of your `json` file is still invalid:

```json
   {
      "name": "tickets",
      "documents": [
        // mock data ...
      ]
    } 
```

In order to have multiple documents inside the collection, you have to have an array of objects inside JSON:

```json
[{"sampleProperty": "foo"}, {"sampleProperty": "bar"}]
```

BTW:

> It would be nice if there was some kind of a warning for cases like mine (maybe an error to the console). If not, this discussion will do the trick 😄

If you enable debug logging, then you can see which exactly collections are populated. In your case would be no collections at all, so I think it could help a bit.

Cheers!

**Reply by listguy** [reply]
Yes I know. 
This `json` file is the one I modified to match your structure (as returned from `readCollectionsFromPath()`) and used it in my workaround.

Thanks again for your patience.

**Reply by pkosiec** [reply]
Oh, okay. So... enjoy using Mongo Seeding then 🙂 In the meantime you can mark the comment https://github.com/pkosiec/mongo-seeding/discussions/156#discussioncomment-510444 as answer for your question 🙂 Cheers!
